src/views/main/components/TopBar.js

```diff
@@ -1,50 +1,110 @@
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { setSearch } from '../../../redux/actions';
 import { FaSearch } from "react-icons/fa";
 import '../../../style/main/components/TopBar.scss';
 import Menu from '../../../components/Menu';
-
+import PropTypes from 'prop-types';
 
 class TopBar extends Component {
 
 	constructor(props) {
 		super(props);
 		this.state = {
-			text: "",
+			searchText: "",
+			searchFocus: false
 		};
 	}
 
 	static propTypes = {
-    text: PropTypes.string
+		onLocationChanged: PropTypes.func
 	}
 
-	render() {
-		const { text } = this.props;
+	componentDidMount() {
+		document.addEventListener('mousedown', this.handleClickOutside);
+	}
 
-		return(
-			<div className="topBar">
+	componentWillUnmount() {
+		document.removeEventListener('mousedown', this.handleClickOutside);
+	}
+
+	handleSearchTextChange = (event) => {
+		var text = event.target.value;
+		this.setState({ searchText: text });
+		if (text.length >= 3) {
+			fetch('http://photon.komoot.de/api/?q=' + text)
+				.then(response => response.json())
+				.then(data => this.setState({ searchData: data }));
+		} else {
+			this.setState({ searchData: null });
+		}
+	}
 
-				<div className="menuHolder">
-					<Menu/>
+	handleSearchFocusOn = (event) => {
+		this.setState({ searchFocus: true });
+	}
+
+	handleSuggestionClick = (feature) => {
+		if (this.props.onLocationChanged) {
+			this.props.onLocationChanged(feature.geometry.coordinates);
+		}
+		this.setState({ searchFocus: false });
+	}
+
+	handleClickOutside = (event) => {
+		if (this.searchAreaRef && !this.searchAreaRef.contains(event.target)) {
+			this.setState({ searchFocus: false });
+		}
+	}
+
+	setSearchAreaRef = (node) => {
+		this.searchAreaRef = node;
+	}
+
+	getAutoCompleteList = () => {
+		const features = this.state.searchData.features;
+		const newFeaturesSize = Math.min(8, features.length);
+		const limitedFeatures = features.slice(0, newFeaturesSize);
+		const listItems = limitedFeatures.map((feature, index) =>
+			<li key={index} onClick={() => this.handleSuggestionClick(feature)}>
+				<span className="suggestionTitle">{feature.properties.name}</span>
+				{feature.properties.street != null &&
+					<span className="suggestionSubtitle">{feature.properties.name}</span>}
+			</li>
+		)
+		return <ol className="autoCompleteBox">
+			{listItems}
+		</ol>
+	}
+
+	render() {
+		return (
+			<div className="topBar">
+				<div className="menuLeftItems">
+					<div style={{ float: 'left' }}>
+						<Menu />
+					</div>
+					{/* logo placeholder */}
+					<span style={{ fontSize: "30px", marginLeft: "20px", marginTop: "4px" }}>Airella</span>
 				</div>
-				<div className="inputHolder">
-					<input className="mainSearch" type="text" />
-					<div className="searchBtn"><FaSearch className="btnSearchIcon"/></div>
+				<div className="searchArea" ref={this.setSearchAreaRef} >
+					<div className="inputHolder" style={{ display: 'flex', height: '100%' }}>
```

Comment:
pliska, nie rób inline stylów bo potem może być ciężko znaleźć takie rzeczy, możesz dodać drugą klasę do elementu i tam to opisać